apps/frontend/src/components/containers/Auth/Otp/Otp.tsx

```diff
@@ -1,47 +1,70 @@
 import React, { useState } from 'react';
-import axios, { type AxiosError } from 'axios';
+import axios from 'axios';
+import { useRouter } from 'next/router';
 
 import OtpView from './Otp.view';
 
 const Otp = () => {
+	const router = useRouter();
+
+	const { otp } = router.query;
+
+	let decodedUrl: string;
+
+	if (typeof otp === 'string') {
+		decodedUrl = decodeURI(otp);
+	}
+
 	const [inputOtpState, setInputOtpState] = useState(new Array(6).fill(''));
 
+	const handleInputPaste = (e: React.ClipboardEvent<HTMLInputElement>) => {
+		const pasteArray = e.clipboardData.getData('text').split('');
+
+		setInputOtpState((prev) => [
+			...prev.map((char, indexMap) => (char = pasteArray[indexMap] ? pasteArray[indexMap] : '')),
+		]);
+	};
+
 	const handleInputChange = (e: React.ChangeEvent<HTMLInputElement>, index: number) => {
+		if (e.nativeEvent instanceof InputEvent && e.nativeEvent.inputType === 'insertFromPaste') return;
+
 		setInputOtpState((prev) => [
 			...prev.map((char, indexMap) => (indexMap === index ? e.target.value : char)),
 		]);
 
-		const nextInput = e.target.nextElementSibling as HTMLInputElement | null;
+		const nextInput = e.target.nextElementSibling;
 
-		if (nextInput) {
+		if (nextInput instanceof HTMLInputElement) {
 			nextInput.focus();
 		}
 	};
 
-	const handleSubmit = (e: React.FormEvent) => {
+	const handleSubmit = async (e: React.FormEvent) => {
 		e.preventDefault();
 
-		axios
-			.post('http://localhost:5000/user/loginotp', {
+		try {
+			const response = await axios.post(`${process.env.NEXT_PUBLIC_BACkEND_URL}/user/loginotp`, {
+				token: decodedUrl,
 				otp: inputOtpState.join(''),
-				email: 'bare1212@gmail.com',
-			})
-			.then(
-				(res) => {
-					console.log(res);
-				},
-				(error: Error | AxiosError) => {
-					if (axios.isAxiosError(error)) {
-						console.error('Error during registration:', error?.response?.data?.message);
-					} else {
-						console.log('An unknown error occurred');
-					}
-				},
-			);
+			});
+
+			console.log(response);
+		} catch (error) {
+			if (axios.isAxiosError(error)) {
+				console.error('Error during registration:', error.response?.data?.message);
+			} else {
+				console.log('An unknown error occurred');
```

Comment:
also that, we told you to display something in the UI for errors

Reply:
you are going to make a list of points from this review

1. don't use `console.log` as error handling.
2. no random spaces between lines,
3. no random `console.logs`.
4. ...... 

add everything you can think about & take points from my comments. next time after you finish writing the code, you will go through all the points and implement those fixes yourself. otherwise I will not review your code ever again.